python/pyspark/pandas/typedef/typehints.py

```diff
@@ -362,8 +362,9 @@ def infer_pd_series_spark_type(
     if dtype == np.dtype("object"):
         if len(pser) == 0 or pser.isnull().all():
             return types.NullType()
-        elif hasattr(pser.iloc[0], "__UDT__"):
-            return pser.iloc[0].__UDT__
+        first_idx = pser.first_valid_index()
+        if first_idx is not None and hasattr(pser.loc[first_idx], "__UDT__"):
+            return pser.loc[first_idx].__UDT__
```

Comment:
Good catch! but using `.loc` sounds dangerous, e.g.:

```py
>>> pser = pd.Series([None,None,3], index=[1,0,1])
>>> i = pser.first_valid_index()
>>> pser.loc[i]
1    NaN
1    3.0
dtype: float64
```

How about using `notnull()`?

```suggestion
        notnull = pser[pser.notnull()]
        if hasattr(notnull.iloc[0], "__UDT__"):
            return notnull.iloc[0].__UDT__
```